src/test/java/org/jenkinsci/plugins/mesos/ConnectionTest.java

```diff
@@ -33,12 +39,28 @@ public void startAgent(TestUtils.JenkinsRule j)
     String mesosUrl = mesosCluster.getMesosUrl();
     MesosApi api = new MesosApi(mesosUrl, "example", "MesosTest");
 
-    MesosSlave agent = api.enqueueAgent().toCompletableFuture().get();
+    MesosCloud cloud = Mockito.mock(MesosCloud.class);
+
+    MesosSlave agent = api.enqueueAgent(cloud, 0.1, 32).toCompletableFuture().get();
 
     // Poll state until we get something.
     while (!agent.isRunning()) {
       Thread.sleep(1000);
       System.out.println("not running yet");
     }
   }
+
+  @Test
+  public void testProvision() throws Exception {
+    LabelAtom label = Mockito.mock(LabelAtom.class);
+
+    MesosCloud cloud = new MesosCloud("mesos", mesosCluster.getMesosUrl(), "jenkinsUrl", "slaveUrl");
+
+    int workload = 3;
+    Collection<NodeProvisioner.PlannedNode> plannedNodes = cloud.provision(label, workload);
+
+    Assert.assertEquals(plannedNodes.size(), workload);
```

Comment:
Hamcrest would give better error messages:
```suggestion
    assertThat(plannedNodes, hasSize(workload));
```

Reply:
added to my TODO list